robot/impl/local_robot_test.go

```diff
@@ -1932,6 +1932,131 @@ func TestConfigPackageReferenceReplacement(t *testing.T) {
 	test.That(t, r.Close(context.Background()), test.ShouldBeNil)
 }
 
+// removeBuiltinServices removes services with a "builtin" name and for testing
+// purposes.
+func removeBuiltinServices(cfg *config.Config) *config.Config {
+	if cfg == nil {
+		return nil
+	}
+
+	var nonBuiltInSvcs []resource.Config
+	for _, svc := range cfg.Services {
+		if svc.Name != resource.DefaultServiceName {
+			nonBuiltInSvcs = append(nonBuiltInSvcs, svc)
+		}
+	}
+	cfg.Services = nonBuiltInSvcs
+	return cfg
+}
+
+func TestConfigMethod(t *testing.T) {
+	// Partially a regression test for RSDK-3177 (internal services and implicit
+	// dependencies should not be returned from Config method).
+
+	ctx := context.Background()
+	logger := golog.NewTestLogger(t)
+
+	// Precompile modules to avoid timeout issues when building takes too long.
+	err := rtestutils.BuildInDir("examples/customresources/demos/complexmodule")
+	test.That(t, err, test.ShouldBeNil)
+
+	r, err := robotimpl.New(context.Background(), &config.Config{}, logger)
+	test.That(t, err, test.ShouldBeNil)
+	defer func() {
+		test.That(t, r.Close(context.Background()), test.ShouldBeNil)
+	}()
+
+	// Assert that Config method returns only built-in services.
+	test.That(t, removeBuiltinServices(r.Config()), test.ShouldResemble, &config.Config{})
```

Comment:
how does this assert that? Wouldn't you need to check that the service length is non 0?

Reply:
good catch.. 😮‍💨 was not awake this morning. asserted that there are three built-in services and they are data manager, motion and sensors.